### ISSUE TITLE
IPv6 toggle or specifying distinct listening address

### ISSUE THREAD

**Issue by krangm** [Jigsaw-Code/outline-server#1601]
Since today, Outline server docker image enforce using IPv6 for some reason.
IPv6 is disabled in my kernel, so enforcing IPv6 does not let server to start with the following error:

```css
ERR Server failed to start. Aborting. err="failed to configure server: unable to create stream listener for [::]:11111: listen tcp [::]:11111: socket: address family not supported by protocol"
```

Currently I have to downgrade to this image, which has no IPv6 enforcing:
`quay.io/outline/shadowbox:v1.9.0`

It would be nice to have an ability to specify desired listening address which will determine protocol to use. This will allow to leave both protocols enabled by default. Or whatever else to toggle IPv6.

**Comment by FAR747** [comment]
How do you downgrade the server version?

**Comment by krangm** [comment]
Recreate server container with same properties, but edit the image tag.From `quay.io/outline/shadowbox:stable` to `quay.io/outline/shadowbox:v1.9.0`On 27 Sep 2024, at 00:24, Valentin Sonin ***@***.***> wrote:﻿
How do you downgrade the server version?

**Comment by NosikovSergey** [comment]
It's work for me too.
I edited script /opt/outline/persisted-state/start_container.sh. Replaced `quay.io/outline/shadowbox:stable` to `quay.io/outline/shadowbox:v1.9.0` how wrote @krangm 
It seems like the latest release broke

**Comment by FAR747** [comment]
I created a bash script

```bash
#!/bin/bash

export SB_IMAGE=quay.io/outline/shadowbox:v1.9.0 # version 
export WATCHTOWER_REFRESH_SECONDS=2592000 # Watchtower refresh 90 days

curl -sSL https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh | bash
```

**Comment by koshelevdk** [comment]
Same thing here. I have ipv6 completely disabled on my machine and fount this issue only after rolling back to v1.9.0 manually.

**Comment by behrouzamiri** [comment]
@NosikovSergey would you mind sharing `/opt/outline/persisted-state/start_container.sh` ? as @FAR747 mentioned, `start_container.sh` is not available in my installation, and I'm curious what options are specified there for running the container?

**Comment by koshelevdk** [comment]
@behrouzamiri  here it is

https://github.com/Jigsaw-Code/outline-server/blob/master/src/server_manager/install_scripts/install_server.sh#L303

Image is controlled by watchtower, that's why it was updated without user interaction
https://github.com/Jigsaw-Code/outline-server/blob/master/src/server_manager/install_scripts/install_server.sh#L367

**Comment by behrouzamiri** [comment]
Thanks @koshelevdk ! that did the trick!
I saw in watchtower logs that it updated shadowbox image 4hours ago, and that led me here.
the part of `install_server.sh` script (`start_shadowbox()` function) that you've pointed me to shed the light and helped me.
it's actually generating the script `start_container.sh` and runs it.
I could simply downgrade the image to 1.9.0 and do a docker run with the required args. it's all good!
also, ran the watchtower with 1 month set as interval, for safety! (thanks to @FAR747 for pointing that out)

**Comment by fortuna** [comment]
This is the culprit:
https://github.com/Jigsaw-Code/outline-ss-server/blob/cb5965fd1a6982b641c640067b55bcdb1ccca8eb/cmd/outline-ss-server/main.go#L215

Probably solvable by using an empty host instead so the address reads as `:$PORT`.

**Comment by sbruens** [comment]
Just a quick update that we believe we have fixed the regression, but have been having trouble reliably reproducing the original problem.

If folks that ran into this issue on the previous version (the image with the bug is `quay.io/outline/shadowbox:v1.10.0`) is able to do a test run with the new release candidate to see if they are still running into the issue on the machines that have IPv6 disabled, that would help give us more confidence before releasing to a wider audience.

New release candidate image:
```
quay.io/outline/shadowbox:v1.10.1-rc1
```

**Comment by krangm** [comment]
```
quay.io/outline/shadowbox:v1.10.1-rc1
```
Works for me. Server starts fine.
IPv6 was disabled by booting with kernel parameter `ipv6.disable=1`

Thank you!

**Comment by est0n14n** [comment]
> ```
> quay.io/outline/shadowbox:v1.10.1-rc1
> ```

Works for me. 
Thank you!

**Comment by sbruens** [comment]
We have released v1.10.1 that has the fix. Please do let us know if you run into further issues with this.

I'll leave this issue open to reflect the feature request to be able toggle it or specify distinct addresses.
